README.md

```diff
@@ -6,13 +6,31 @@ Download your spotify playlists using simple python script
 * Adds metadata to the downloaded songs (title, artist and album)
 * Download only the songs you don't have
 
+
+### Installation
+If you already have [Python](http://www.python.org/) on your system you can install the library simply by downloading the distribution, unpack it and install in the usual fashion: (May require root priveleges)
+
+    python setup.py install
+
 ### Downloading playlist
+>`$ cd spotify_download`
+> (For usage: `python download.py -h`)
+* Use [download.py](spotify_download/download.py) to download all songs from the playlist.
+
+1. Download through CSV File
+    * Convert your spotify playlists to csv from [here](http://joellehman.com/playlist/) (Thanks to [Joel Lehman](https://github.com/jal278))
 
-* Install dependencies: `pip install youtube_dl eyed3 unidecode` (May require root priveleges)
-* Convert your spotify playlists to csv from [here](http://joellehman.com/playlist/) (Thanks to [Joel Lehman](https://github.com/jal278))
-* Use [download_from_csv.py](download_from_csv.py) to download all songs from the playlist. (For usage: `python download_from_csv.py -h`)
+2. Download through User Login
+    * Login to your spotify account and Create your [Application](https://developer.spotify.com/my-applications/#!/applications)
+        * Set Redirect URIs = `http://localhost/` in your application settings and save it.
```

Comment:
Actually, User authentication requires interaction with the web browser. Once the user enters credentials and give authorization, the user will be redirected to an URL by default and if we do not provide any 'SPOTIPY_REDIRECT_URI' then it will lead to error `Invalid Redirect URI`. 

Reference :- [Link](https://spotipy.readthedocs.io/en/latest/#authorized-requests)